db/src/id.rs

```diff
@@ -136,38 +134,34 @@ impl<'a> Iterator for IdsIter<'a> {
 }
 
 pub struct LiveSet {
-    filter: CuckooFilter<FnvHasher>,
-    distinct: usize,
+    filter: FnvHashMap<Id, u32>,
```

Comment:
If we assume that we will not have more than let's say a million blocks, we could also use an u32 as key here and use 32 bits of the hash of the id. Then you would occasionally have a collision, but you are guaranteed never to underestimate your count, so the consequence would be just occasionally a block being accidentally retained. Might be a reasonable tradeoff for smaller filter size.

However, I think u64 -> u64 is the way to go here. It will be rounded up to a multiple of 8 bytes anyway on 64bit arch.

Reply:
good points